thunx-model/src/main/java/eu/xenit/contentcloud/thunx/predicates/model/ResolvedExpression.java

```diff
@@ -0,0 +1,14 @@
+package eu.xenit.contentcloud.thunx.predicates.model;
+
+import java.util.Optional;
+
+public interface ResolvedExpression<T> extends ThunkExpression<T> {
+    T getResult();
+
+    static <R> Optional<R> maybeResult(ThunkExpression<R> expression) {
```

Comment:
Given this api, I expect any `Scalar<R>` argument to directly resolve to a non-empty `Optional<R>`, without any detour via `simplify()` ?

Reply:
hmm, maybe the signature should be `T getValue()` and at that point `Scalar<T>` could implement `ResolvedExpression<T>` 
(bonus: you can simply remove ResolvedExpressionImpl<T>` ?)

Reply:
The reason why I introduced a separate interface/class is for the benefit of the `ThunkExpressionVisitor`.

Since it is a separate type of expression, we can call a separate method on the visitor that prevents completely simplified expressions from being JSON serialized or embedded in QueryDSL contexts: https://github.com/xenit-eu/thunx/pull/11/files#diff-7888bb2a7b123199970e4fddd0cf6fe94ab797e38b93333100e4d6e5026c5194R81